src/dune_rules/pkg_rules.ml

```diff
@@ -351,7 +351,22 @@ module Pkg = struct
         ]
   ;;
 
-  let deps_closure t = top_closure t.depends
+  let deps_closure t =
+    let closure = top_closure t.depends in
+    List.map closure ~f:(fun pkg -> pkg.info.name, pkg)
+    |> Dune_pkg.Package_name.Map.of_list_multi
+    |> Dune_pkg.Package_name.Map.iter ~f:(function
+      | _ :: _ :: _ as multiple ->
```

Comment:
Why does a list with >= 3 elements let us know we have a duplicate? Adding a comment to explain that logic would help me (as an admittedly new) reader.

Reply:
Ah yeah! My bad. 

So, IIUC, what we are checking here is that each list is a singleton, and if there is more than one element, it will have duplicates?

If so, I think this would be clearer as something like

```suggestion
      | [_] - () 
      | multiple -> (* more than one of the dependency found *)
```

unless I am missing something?

Reply:
I ended up deleting this code as it was leftover debugging code.